src/sagas/nanoContract.js

```diff
@@ -109,8 +119,142 @@ export function* registerNanoContract({ payload }) {
   yield put(nanoContractRegisterSuccess({ entryKey: ncEntryKey, entryValue: ncEntryValue }));
 }
 
+/**
+ * @typedef {Object} RawNcTxHistory
+ * @property {string} hash
+ * @property {Object[]} inputs
+ * @property {Object[]} outputs
+ * @property {string} nc_args
+ * @property {string} nc_id
+ * @property {string} nc_method
+ * @property {string} nc_pubkey
+ * @property {number} nonce
+ * @property {Object[]} parents
+ * @property {number} signal_bits
+ * @property {Date} timestamp
+ * @property {string[]} tokens
+ * @property {number} version
+ * @property {number} weight
+ *
+ * @typedef {Object} RawNcTxHistoryResponse
+ * @property {boolean} success
+ * @property {number|null} after
+ * @property {RawNcTxHistory} history
+ */
+
+/**
+ * @typedef {Object} NcTxHistory - Nano Contract's Transactions History
+ * @property {string} txId - transaction hash
+ * @property {number} timestamp
+ * @property {string} tokens - list of token's uid
+ * @property {{[uid: string]: Object}} balance - balance per token
+ * @property {boolean} isVoided - flag for transaction void
+ * @property {string} ncId - ID of Nano Contract
+ * @property {string} ncMethod - method called on Nano Contract
+ * @property {string} blueprintId - id of the Blueprint instantiated by Nano Contract
+ * @property {string} caller - address of the caller
+ * @property {'mine'|'nc'|'oracle'|'wallet'} callerOrigin - caller's origin
+ */
+
+/**
+ * @param {RawNcTxHistory} each
+ * @returns {NcTxHistory}
+ */
+function toNcTxHistory(each) {
+  return {
+    txId: each.hash,
+    timestamp: each.timestamp,
+    tokens: each.tokens,
+    ncId: each.nc_id,
+    ncMethod: each.nc_method,
+    // balance: getTxBalance.bind(transactionUtils)(each),
+    // isVoided: false,
+    // blueprintId: each.blueprintId,
+    // caller: getCaller(each),
+    // callerOrigin: getCallerOrigin(each),
+  };
+}
+
+/**
+ * Fetch history from Nano Contract wallet's API.
+ * @param {string} ncId Nano Contract ID
+ * @param {number} count Maximum quantity of history items
+ * @param {string} after Transaction hash to start to get items
+ */
+export async function fetchHistory(ncId, count, after) {
+  /**
+   * @type {RawNcTxHistoryResponse} response
+   */
+  const response = await ncApi.getNanoContractHistory(ncId, count, after);
+  const { success, history } = response;
```

Comment:
blocking: `ncApi.getNanoContractHistory` will [throw an error](https://github.com/HathorNetwork/hathor-wallet-lib/blob/af7815dd98b2c6d7f4da77d5886e28c742f127d9/src/api/nano.ts#L78) if the request fails